File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
using namespace std;

int main(void)
{
	ios_base::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);

	int N;
	cin >> N;

	stack<int> stk;
	string str;

	for (int i = 0; i < N; i++) {
		cin >> str;
		if (str == "push") {
			int k;
			cin >> k;
			stk.push(k);
		}

		else if (str == "pop") {
			if (stk.empty())
				cout << "-1\n";
			else {
				cout << stk.top() << "\n";
				stk.pop();
			}
		}

		else if (str == "size")
			cout << stk.size() << "\n";

		else if (str == "empty") {
			if (stk.empty())
				cout << "1\n";
			else
				cout << "0\n";
		}

		else if (str == "top") {
			if (stk.empty())
				cout << "-1\n";
			else
				cout << stk.top() << "\n";
		}
	}

	return 0;
}

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuffer sb = new StringBuffer();
        int N = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();
        while (N-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String order = st.nextToken();
            if (order.equals("push")) {
                stack.push(Integer.parseInt(st.nextToken()));
            } else if (order.equals("pop")) {
                sb.append(stack.isEmpty() ? -1 : stack.pop()).append("\n");
            } else if (order.equals("size")) {
                sb.append(stack.size()).append("\n");
            } else if (order.equals("empty")) {
                sb.append(stack.isEmpty() ? 1 : 0).append("\n");
            } else if (order.equals("top")) {
                sb.append(stack.isEmpty() ? -1 : stack.peek()).append("\n");
            }
        }
        System.out.println(sb);
    }
}